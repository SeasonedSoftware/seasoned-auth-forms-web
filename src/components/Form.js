import React from 'react'
import PropTypes from 'prop-types'
import { Grid } from '@material-ui/core'
import { Form as FormikForm } from 'formik'

import Button from './Button'
import Loading from './Loading'
import Error from './Error'

const AuthForm = props => {
  const { form, children } = props
  const { submitting, submitError, renderButton, buttonText } = props
  const disabled = submitting || Object.entries(form.errors).length > 0

  return (
    <FormikForm>
      {children}
      {submitError && <Error>{submitError}</Error>}
      {submitting && <Loading style={{ alignSelf: 'center' }} />}
      {renderButton({ text: buttonText, type: 'submit', disabled })}
    </FormikForm>
  )
}

AuthForm.defaultProps = {
  renderButton: buttonProps => (
    <Grid container justify="center">
      <Button {...buttonProps} />
    </Grid>
  ),
}

AuthForm.propTypes = {
  buttonText: PropTypes.string,
  form: PropTypes.object,
  submitError: PropTypes.string,
  submitting: PropTypes.bool,
  renderButton: PropTypes.func,
}

export default AuthForm
