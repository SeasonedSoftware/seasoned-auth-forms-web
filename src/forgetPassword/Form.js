import React from 'react'

import AuthForm from '../components/Form'
import EmailField from '../components/EmailField'

const ForgetPassword = ({ language, ...props }) => {
  return (
    <AuthForm {...props}>
      <EmailField {...props.form} language={language} />
    </AuthForm>
  )
}

ForgetPassword.defaultProps = {
  buttonText: 'Send email with instructions to reset my password',
  language: 'en-US',
}
export default ForgetPassword
