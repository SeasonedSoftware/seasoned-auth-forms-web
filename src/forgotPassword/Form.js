import React from 'react'

import AuthForm from '../components/Form'
import EmailField from '../components/EmailField'

const ForgotPassword = ({ language, ...props }) => {
  return (
    <AuthForm {...props}>
      <EmailField {...props.form} language={language} />
    </AuthForm>
  )
}

ForgotPassword.defaultProps = {
  buttonText: 'Send email with instructions to reset my password',
  language: 'en-US',
}
export default ForgotPassword
