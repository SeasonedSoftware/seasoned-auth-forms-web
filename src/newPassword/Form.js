import React from 'react'

import AuthForm from '../components/Form'
import PasswordField from '../components/PasswordField'

const ForgetPassword = ({ language, ...props }) => {
  return (
    <AuthForm {...props}>
      <PasswordField {...props.form} language={language} />
      <PasswordField {...props.form} language={language} confirmation />
    </AuthForm>
  )
}

ForgetPassword.defaultProps = {
  buttonText: 'Set',
  language: 'en-US',
}
export default ForgetPassword
