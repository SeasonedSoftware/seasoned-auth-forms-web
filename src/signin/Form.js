import React from 'react'
import PropTypes from 'prop-types'
import { Form } from 'formik'
import { Grid } from '@material-ui/core'

import Field from '../components/Field'
import Button from '../components/Button'
import Loading from '../components/Loading'
import Error from '../components/Error'

const SignInForm = ({ form, submitting, submitError, ...props }) => (
  <Grid
    container
    alignContent={'center'}
    justify={'center'}
    style={{ margin: '30px 0' }}
  >
    <Form>
      <Field
        {...form}
        key={'email'}
        name={'email'}
        label={'Email'}
        type={'email'}
        value={form.values.email}
        fullWidth
      />
      <Field
        {...form}
        key={'password'}
        name={'password'}
        label={'Password'}
        type={'password'}
        value={form.values.password}
        fullWidth
      />
      {submitError && <Error>{submitError}</Error>}
      {submitting && <Loading style={{ alignSelf: 'center' }} />}
      <Grid container justify="center">
        <Button
          text={props.buttonText}
          disabled={submitting || Object.entries(form.errors).length > 0}
        />
      </Grid>
    </Form>
  </Grid>
)

SignInForm.defaultProps = {
  buttonText: 'Login',
}

SignInForm.propTypes = {
  submitting: PropTypes.bool.isRequired,
  submitError: PropTypes.string.isRequired,
}

export default SignInForm
