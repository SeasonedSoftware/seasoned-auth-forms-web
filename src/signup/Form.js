import React from 'react'
import PropTypes from 'prop-types'
import { Form } from 'formik'
import { Grid } from '@material-ui/core'

import Field from '../components/Field'
import Button from '../components/Button'
import Loading from '../components/Loading'
import Error from '../components/Error'

const SignUpForm = ({ form, submitting, submitError, ...props }) => (
  <Form>
    <Field
      {...form}
      key={'signUpEmail'}
      name={'email'}
      label={'Email'}
      type={'email'}
      value={form.values.email}
      fullWidth
    />
    <Field
      {...form}
      key={'signUpPassword'}
      name={'password'}
      label={'Password'}
      type={'password'}
      value={form.values.password}
      fullWidth
    />
    <Field
      {...form}
      key={'signUpPasswordConfirmation'}
      name={'passwordConfirmation'}
      label={'Password Confirmation'}
      type={'password'}
      value={form.values.passwordConfirmation}
      fullWidth
    />
    {submitError && <Error>{submitError}</Error>}
    {submitting && <Loading style={{ alignSelf: 'center' }} />}
    <Grid container justify="center">
      <Button
        text={props.buttonText}
        disabled={submitting || Object.entries(form.errors).length > 0}
      />
    </Grid>
  </Form>
)

SignUpForm.defaultProps = {
  buttonText: 'SignUp',
}

export default SignUpForm
