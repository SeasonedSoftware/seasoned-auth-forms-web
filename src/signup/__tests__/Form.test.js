import React from 'react'
import renderer from 'react-test-renderer'

import Form from '../Form'

jest.mock('formik', () => ({
  Form: ({ children, ...props }) => <div {...props}>Form - {children}</div>,
}))

jest.mock('../../components/Field', () => props => <div {...props}>Field</div>)
jest.mock('../../components/Button', () => props => (
  <div {...props}>Button</div>
))
jest.mock('../../components/Loading', () => props => (
  <div {...props}>Loading</div>
))
jest.mock('../../components/Error', () => props => <div {...props}>Error</div>)

it('renders correctly', () => {
  const props = {
    form: {
      values: {},
      errors: {},
    },
    submitError: '',
    submitting: false,
  }
  const tree = renderer.create(<Form {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})

describe('with validation errors', () => {
  it('renders correctly', () => {
    const props = {
      form: {
        values: {},
        errors: { email: 'is not a valid email' },
      },
      submitError: '',
      submitting: false,
    }
    const tree = renderer.create(<Form {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('while submitting', () => {
  it('renders correctly', () => {
    const props = {
      form: {
        values: {},
        errors: {},
      },
      submitError: '',
      submitting: true,
    }
    const tree = renderer.create(<Form {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('with submit errors', () => {
  it('renders correctly', () => {
    const props = {
      form: {
        values: {},
        errors: {},
      },
      submitError: 'Fail',
      submitting: false,
    }
    const tree = renderer.create(<Form {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
