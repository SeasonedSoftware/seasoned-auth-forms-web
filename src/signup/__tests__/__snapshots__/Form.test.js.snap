// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  Form - 
  <div
    errors={Object {}}
    fullWidth={true}
    label="Email"
    name="email"
    type="email"
    values={Object {}}
  >
    Field
  </div>
  <div
    errors={Object {}}
    fullWidth={true}
    label="Password"
    name="password"
    type="password"
    values={Object {}}
  >
    Field
  </div>
  <div
    errors={Object {}}
    fullWidth={true}
    label="Password Confirmation"
    name="passwordConfirmation"
    type="password"
    values={Object {}}
  >
    Field
  </div>
  
  <div
    className="MuiGrid-container-1 MuiGrid-justify-xs-center-18"
  >
    <div
      disabled={false}
      text="SignUp"
    >
      Button
    </div>
  </div>
</div>
`;

exports[`while submitting renders correctly 1`] = `
<div>
  Form - 
  <div
    errors={Object {}}
    fullWidth={true}
    label="Email"
    name="email"
    type="email"
    values={Object {}}
  >
    Field
  </div>
  <div
    errors={Object {}}
    fullWidth={true}
    label="Password"
    name="password"
    type="password"
    values={Object {}}
  >
    Field
  </div>
  <div
    errors={Object {}}
    fullWidth={true}
    label="Password Confirmation"
    name="passwordConfirmation"
    type="password"
    values={Object {}}
  >
    Field
  </div>
  
  <div
    style={
      Object {
        "alignSelf": "center",
      }
    }
  >
    Loading
  </div>
  <div
    className="MuiGrid-container-1 MuiGrid-justify-xs-center-18"
  >
    <div
      disabled={true}
      text="SignUp"
    >
      Button
    </div>
  </div>
</div>
`;

exports[`with submit errors renders correctly 1`] = `
<div>
  Form - 
  <div
    errors={Object {}}
    fullWidth={true}
    label="Email"
    name="email"
    type="email"
    values={Object {}}
  >
    Field
  </div>
  <div
    errors={Object {}}
    fullWidth={true}
    label="Password"
    name="password"
    type="password"
    values={Object {}}
  >
    Field
  </div>
  <div
    errors={Object {}}
    fullWidth={true}
    label="Password Confirmation"
    name="passwordConfirmation"
    type="password"
    values={Object {}}
  >
    Field
  </div>
  <div>
    Error
  </div>
  <div
    className="MuiGrid-container-1 MuiGrid-justify-xs-center-18"
  >
    <div
      disabled={false}
      text="SignUp"
    >
      Button
    </div>
  </div>
</div>
`;

exports[`with validation errors renders correctly 1`] = `
<div>
  Form - 
  <div
    errors={
      Object {
        "email": "is not a valid email",
      }
    }
    fullWidth={true}
    label="Email"
    name="email"
    type="email"
    values={Object {}}
  >
    Field
  </div>
  <div
    errors={
      Object {
        "email": "is not a valid email",
      }
    }
    fullWidth={true}
    label="Password"
    name="password"
    type="password"
    values={Object {}}
  >
    Field
  </div>
  <div
    errors={
      Object {
        "email": "is not a valid email",
      }
    }
    fullWidth={true}
    label="Password Confirmation"
    name="passwordConfirmation"
    type="password"
    values={Object {}}
  >
    Field
  </div>
  
  <div
    className="MuiGrid-container-1 MuiGrid-justify-xs-center-18"
  >
    <div
      disabled={true}
      text="SignUp"
    >
      Button
    </div>
  </div>
</div>
`;
